src/ai/backend/common/docker.py

```diff
@@ -279,53 +272,63 @@ async def login(
     raise RuntimeError(f"authentication for docker registry {registry_url} failed")
 
 
-async def get_known_registries(etcd: AsyncEtcd) -> Mapping[str, yarl.URL]:
-    data = await etcd.get_prefix("config/docker/registry/")
-    results: MutableMapping[str, yarl.URL] = {}
-    for key, value in data.items():
-        name = etcd_unquote(key)
-        if isinstance(value, str):
-            results[name] = yarl.URL(value)
-        elif isinstance(value, Mapping):
-            assert isinstance(value[""], str)
-            results[name] = yarl.URL(value[""])
-    return results
+def is_default_registry(registry_name: str, project: str) -> bool:
+    return project == "" and registry_name == default_registry
 
 
-def is_known_registry(
-    val: str,
-    known_registries: Union[Mapping[str, Any], Sequence[str]] | None = None,
-):
-    if val == default_registry:
-        return True
-    if known_registries is not None and val in known_registries:
-        return True
+def is_dockerhub_library_registry(
+    project: str,
+    known_registries: Optional[Mapping[str, Mapping[str, Any]] | Sequence[str]],
+) -> bool:
+    if project == "":
+        if isinstance(known_registries, list) and "index.docker.io" in known_registries:
+            return True
+        if isinstance(known_registries, Mapping):
+            return "index.docker.io" in known_registries["library"]
```

Comment:
is it sure that "library" key is always in `known_registries` dictionary? if not, we should handle the case of not having "library" key

Reply:
No, key name could be different with `library`, so I think it should be 

```py
        if isinstance(known_registries, Mapping):
            return "index.docker.io" in known_registries.get("library")
```